src/kbmod/analysis_utils.py

```diff
@@ -835,39 +685,30 @@ def _clipped_average(self, psi_curve, phi_curve, index, num_clipped=5, n_sigma=4
             new_lh = kb.calculate_likelihood_psi_phi(psi_curve[max_lh_index], phi_curve[max_lh_index])
         return (index, max_lh_index, new_lh)
 
-    def apply_kalman_filter(self, old_results, filter_params):
+    def apply_kalman_filter(self, result_list, filter_params):
         """
         This function applies a kalman filter to the results of a KBMOD search
-            INPUT-
-                old_results : dictionary
-                    Dictionary containing values from trajectories. When input,
-                    it should have at least 'psi_curves', 'phi_curves', and
-                    'results'. These are populated in Interface.load_results().
-                filter_params : dictionary
-                    A dictionary of additional filtering parameters.
-            OUTPUT-
-                keep_idx_results : list
-                    list of tuples containing the index of a results, the
-                    indices of the passing values in the lightcurve, and the
-                    new likelihood for the lightcurve.
+
+        Arguments:
+            result_list (`ResultList`): The values from trajectories. This data gets
+                modified directly by the filtering.
+            filter_params (dict): A dictionary of additional filtering parameters.
         """
         print("Applying Kalman Filtering")
         start_time = time.time()
-        # Make copies of the values in 'old_results' and create a new dict
-        psi_curves = np.copy(old_results["psi_curves"])
-        phi_curves = np.copy(old_results["phi_curves"])
 
-        keep_idx_results = kb.kalman_filtered_indices(psi_curves.tolist(), phi_curves.tolist())
+        for row in result_list.results:
+            keep_idx_results = kb.kalman_filtered_indices([row.psi_curve], [row.phi_curve])
```

Comment:
Is this correct because it's confusing? Old code had psi and phi as numpy arrays, so a `psi.tolist()` returns the same thing, just as a straight python `list`. `ResultRow.set_psi_phi` sets them as `lists`. So a `[psi/phi]` would be a list of lists in my mind? 
This seemingly works, because nothing breaks when I run it, but its confusing me.

Reply:
I think it is correct, but less efficient. The previous code took a list of lists with one list 'row' for each result. In the new code we put the loop outside. Since the code is still expecting a list of lists, we feed them in one at a time. It's not a great solution. Let me know if you see a better way to do this.

Reply:
if you wanted to, you could just create a binding for [kalmanFilterIndex](https://github.com/dirac-institute/kbmod/blob/main/src/kbmod/search/Filtering.cpp#L224) and call that directly instead! I also know we're thinking about getting rid of kalman filtering entirely so it might not be worth the effort 😭

Reply:
I think we should prune the filtering options, so let's leave it as-is until we decide which filtering options to keep.